### ISSUE TITLE
screenshots are not found in the serenity report

### ISSUE THREAD

**Issue by Prasannajnaeyulu** [serenity-bdd/serenity-cucumber-starter#2]
I am using this project. But, the screenshots are not found in the aggregated serenity report. Could you please help me out!!

I tried..
serenity.take.screenshots = AFTER_EACH_STEP and also
serenity.take.screenshots = FOR_FAILURES

But nothing works out for me!!!

**Comment by Prasannajnaeyulu** [comment]
you can see in the attached screenshot, I haven't seen the screenshots in the reports when I configured
this serenity.take.screenshots = AFTER_EACH_STEP in serenity.properties file. could you please help me with the right  way to configure in order to embed screenshots into the report?

**Comment by wakaleo** [comment]
This is not a web test, it is just a demo illustrating a Cucumber scenario. If you want a wettest, look at https://github.com/serenity-bdd/journey-planner

**Comment by Prasannajnaeyulu** [comment]
It seems to be all of our actions should implement interactions interface for the screenshots to work. As in the following example....
**public class PickFirstSuggestion implements Interaction**

But In my project, I didn't. I am using serenity just for running with cucumberWithSerentiy runner class and for reporting using serenitymaven plugin. For Implementation of the tests, I am using cucumber core library.  In this project structure, Didn't I get the screenshots into reports? By having calls to either of serenity.takescreenshot method or scenario.embed method inside @After hook method of cucumber.

**Comment by wakaleo** [comment]
No, to get screenshots in your reports you need to let Serenity handle the WebDriver lifecycle

**Comment by Prasannajnaeyulu** [comment]
That's an Instant reply. Much appreciated. **Serenity handle the WebDriver lifecycle** can you please point me to the link that describes more on this?

**Comment by Prasannajnaeyulu** [comment]
hmmmm!!! I tried using a listener as described here... 
https://schenkweb.nl/2017/07/11/serenity-bdd-register-a-custom-listener/

But, the listener is not at all calling. It looks like this custom listener will only work for SerenityRunner but not with cucumberWithSerenity Runner class

**Comment by wakaleo** [comment]
You need to run your tests using the CucumberWithSerenity runner and structure your tests using either Serenity step libraries (@Steps and @Step) or Screenplay classes.

**Comment by Prasannajnaeyulu** [comment]
Yeah, This is what I thought after seeing multiple examples. But, does the step listener also warrants this?

**Comment by wakaleo** [comment]
The step listeners aren't really designed for this purpose.

**Comment by Prasannajnaeyulu** [comment]
Ok. I structured the steps using @step annotation as shown below... 	
@Step
@Screenshots(onlyOnFailures=true)
public String getText(By locator) {
.........
}

@Step
@Screenshots(onlyOnFailures=true)
public boolean isElementVisible(By locator) {
......................
}

But, Still I am not getting the screenshots on failure. I am using cuucmberWithSerenity for running. Do we need to configure something somewhere to get the screenshots on to reports. Can you please help me out with the configuration. Right now, web driver is managing by my framework but not by serenity. As I told you, I am using serenity only for runner and reporting. For the screenshots purpose only, I have added @step to the step. definition actual implementation methods as said above.

**Comment by wakaleo** [comment]
This won't work. If you want to benefit from the Serenity screenshot reporting, you need to use the Serenity framework as it is intended; it is not designed to be mixed with custom frameworks.

**Comment by ftchambo** [comment]
Hello,
I have a Problem  with taking Screenshots while using the SerenityParameterizedRunner
I write a class which inherit from the SerenityParameterizedRunner because i needed to dynamicaly handle the place where to look for the CSV-files.
`public class MYCSVRunner extends SerenityParameterizedRunner { }`

`Serenity.take.screenshots = BEFORE_AND_AFTER_EACH_STEP`

<img width="2560" alt="bildschirmfoto 2018-03-07 um 17 03 34" src="https://user-images.githubusercontent.com/11678490/37103042-9b0fce92-2229-11e8-8ed0-5d26c7633976.png">

**Comment by AndreiAntipov** [comment]
Serenity should manage your webdriver to take screenshots.
I've extend my DriverFactory - extends ThucydidesWebDriverSupport, then for example:
ChromeDriver driver = new ChromeDriver(options);
useDriver(driver);
return driver;
Now I have screenshots in reports using mixed framework.
I hope this will help.

**Comment by ManikandanSekhar** [comment]
Hi AndreiAntipov,
I am also looking to get the screen shot with serenity report while using CucumberwithSerenity . . I tried the suggestion you had provided earlier as below

ChromeDriver driver = new ChromeDriver(options);
useDriver(driver);
return driver;

unfortunately I did not get the screenshot. Can you please share your project it would be really helpful.

**Comment by marcandreuf** [comment]
Solved! 

Hello I had this issue of missing screenshots and I found that the maven-failsafe-plugin needs to have the this configuration.
```
<properties>
....
     <webdriver.driver>firefox</webdriver.driver>
....
</properties>
.....
           <!-- Integration tests reports plugin -->
            <plugin>
                <artifactId>maven-failsafe-plugin</artifactId>
                <version>2.19.1</version>
                <configuration>
                    <includes>
                        <include>**/*Tests.java</include>
                    </includes>
                    <argLine>-Xmx512m</argLine>
                    **<systemPropertyVariables>
                        <webdriver.driver>${webdriver.driver}</webdriver.driver>
                    </systemPropertyVariables>**
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>integration-test</goal>
                            <goal>verify</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
.....

```
This is explained at the end of the official documentation section [4.2. Building Serenity projects in Maven](http://thucydides.info/docs/serenity-staging/)

I was badly assuming that if I pass the webdriver.driver via system property as -Dwebdriver.driver, then I would not need this configuration in the maven-failsafe-plugin, but somehow the configuration is still necessary to initiate the framework properly. The system property can be added to the command line to choose another driver anyway even with this configuration.
No all scenarios that use screenplay pattern have their screenshots.

**Comment by sdas020** [comment]
Screen shot is working for me ..Thanks for sharing your ideas @ManikandanSekhar

**Comment by anhphan176** [comment]
@sdas020 Can you please share it?

**Comment by rajaaryan4** [comment]
@sdas020  Can you please share your project where the screenshot is populated in the serenity report?. I am unable to get the screenshot in the serenity report, Thanks in advance

**Comment by rohithQA** [comment]
Am unable to get the screenshot for serenity POM project 
I have added serenity.take.screenshots=BEFORE_EACH_STEP in serenity properties
even though am unable to get it on mentioned propertie
This is my serenity.properties file :
webdriver.driver = chrome
#webdriver.driver = Microsoft Edge
webdriver.driver = firefox
webdriver.testdata.file = src/test/resources/datafile/Book2.csv
serenity.browser.maximized=true
serenity.take.screenshots=BEFORE_AND_AFTER_EACH_STEP
and this is my dependencies 
<project xmlns="http://maven.apache.org/POM/4.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>com.serenityPOM</groupId>
	<artifactId>curaHealthManagement</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<properties>
		<serenity.version>2.0.81</serenity.version>
		<serenity.maven.version>2.0.81</serenity.maven.version>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<encoding>UTF-8</encoding>
	</properties>


	<dependencies>


		<dependency>
			<groupId>net.serenity-bdd</groupId>
			<artifactId>serenity-core</artifactId>
			<version>${serenity.version}</version>

		</dependency>
		<dependency>
			<groupId>net.serenity-bdd</groupId>
			<artifactId>serenity-junit</artifactId>
			<version>${serenity.version}</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>net.serenity-bdd</groupId>
			<artifactId>serenity-screenplay</artifactId>
			<version>${serenity.version}</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>net.serenity-bdd</groupId>
			<artifactId>serenity-screenplay-webdriver</artifactId>
			<version>${serenity.version}</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>net.serenity-bdd</groupId>
			<artifactId>serenity-ensure</artifactId>
			<version>${serenity.version}</version>
			<scope>test</scope>
		</dependency>

		

		
		<dependency>
			<groupId>org.apache.poi</groupId>
			<artifactId>poi</artifactId>
			<version>4.1.1</version>
		</dependency>

		
		<dependency>
			<groupId>org.apache.poi</groupId>
			<artifactId>poi-ooxml</artifactId>
			<version>4.1.1</version>
		</dependency>

		
		<dependency>
			<groupId>dom4j</groupId>
			<artifactId>dom4j</artifactId>
			<version>1.6.1</version>
		</dependency>

		
		<dependency>
			<groupId>org.apache.xmlbeans</groupId>
			<artifactId>xmlbeans</artifactId>
			<version>3.1.0</version>
		</dependency>

		
		<dependency>
			<groupId>org.apache.poi</groupId>
			<artifactId>poi-ooxml-schemas</artifactId>
			<version>4.1.1</version>
		</dependency>

		


	</dependencies>


	<build>
		<plugins>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-surefire-plugin</artifactId>
				<version>2.22.1</version>
				<configuration>
					<skip>true</skip>
				</configuration>
			</plugin>
			<plugin>
				<artifactId>maven-failsafe-plugin</artifactId>
				<version>2.22.1</version>
				<configuration>
					<includes>


						
						
						<include>**/testcases/*.java</include>

					</includes>

					
				</configuration>
				<executions>
					<execution>
						<goals>
							<goal>integration-test</goal>
							<goal>verify</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-compiler-plugin</artifactId>
				<version>3.7.0</version>
				<configuration>
					<source>1.8</source>
					<target>1.8</target>
				</configuration>
			</plugin>
			<plugin>
				<groupId>net.serenity-bdd.maven.plugins</groupId>
				<artifactId>serenity-maven-plugin</artifactId>
				<version>${serenity.version}</version>
				<configuration>
					<tags>${tags}</tags>
				</configuration>
				<executions>
					<execution>
						<id>serenity-reports</id>
						<phase>post-integration-test</phase>
						<goals>
							<goal>aggregate</goal>
							<goal>check</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
		</plugins>
	</build>

</project>
